Zend/tests/match/031.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+Test warning on duplicate match conditions
+--FILE--
+<?php
+
+match (1) {
+    1, 2, 3, 4, 5, 5 => null,
```

Comment:
This is inconsistent in some ways:
- This will warn about `1, 1 => null` but not `1, 1, '1' => null`
- This may depend on the plaform being used, e.g. `match ($otherOs) { \PHP_OS, 'Linux' => ...}` will warn on some platforms but not others
- I think this may behave differently based on whether a native module or a polyfill is used, or whether `opcache.preload` is used.

Also, in languages such as C/Rust, the warning would get emitted at compile time, but it wouldn't affect the runtime behavior of the program

Reply:
> This will warn about 1, 1 => null but not 1, 1, '1' => null

Yeah. I'm not sure if it's better to warn sometimes than never at all. Also, code like this is technically possible.

```php
switch ($x) {
    1 => ...,
    $x = 1 => ...,
    1 => ...,
}
```

So a warning here might even be wrong.

> This may depend on the plaform being used, e.g. match ($otherOs) { \PHP_OS, 'Linux' => ...} will warn on some platforms but not others

This is a good point.

> I think the warning would get emitted at compile time and could be suppressed, but it wouldn't affect the runtime behavior of the program

This is a compile time error as well. It doesn't affect runtime.

Reply:
Your points are all valid, I removed the warning for now 🙂